assembl/static2/js/app/components/debate/brightMirror/fictionBody.jsx

```diff
@@ -1,32 +1,188 @@
+const noContentMessage: string = 'no content specified';
+
+class FictionBody extends React.Component<FictionBodyProps, State> {
+  fictionBodyView: ?HTMLElement;
+
+  constructor(props: FictionBodyProps) {
+    super(props);
+    this.state = {
+      displayCommentAnchor: false,
+      displaySubmitBox: false,
+      commentAnchorPosition: { x: 0, y: 0 },
+      commentBadgeDynamicPosition: { x: 0, y: 0 },
+      commentBadgeFixedPosition: { x: 0, y: 0 },
+      commentBadgePositionInit: false
+    };
+  }
+
+  componentDidMount() {
+    const { commentBadgePositionInit } = this.state;
+    // Set badge to right of body right after rendering to get correct coordinates
+    if (!commentBadgePositionInit) this.setCommentBadgeFixedPosition();
+  }
+
+  setCommentBadgeFixedPosition = () => {
+    const { dbId } = this.props;
+    // $FlowFixMe because element may be null
+    const body = document.getElementById(`message-body-local:Content/${dbId}`).getBoundingClientRect();
+    this.setState({
+      commentBadgeFixedPosition: { x: body.right, y: body.top },
+      commentBadgePositionInit: true
+    });
+  };
+
+  setCommentBadgeToExtractPosition = (extract: FictionExtractFragment) => {
+    const extractElement = document.getElementById(`${extract.id}`);
+    if (extractElement && this.fictionBodyView) {
+      // Scroll extract in middle of page
+      const elementRect = extractElement.getBoundingClientRect();
+      const absoluteElementTop = elementRect.top + window.pageYOffset;
+      const middle = absoluteElementTop - window.innerHeight / 2;
+      window.scrollTo(0, middle);
+      // $FlowFixMe this.fictionBodyView may be null
+      const nexPositionY = extractElement.getBoundingClientRect().top - this.fictionBodyView.getBoundingClientRect().top;
+      this.setState({
+        commentBadgeDynamicPosition: {
+          // $FlowFixMe this.fictionBodyView may be null
+          x: this.fictionBodyView.offsetLeft + this.fictionBodyView.clientWidth,
+          y: nexPositionY - COMMENT_DYNAMIC_OFFSET
+        }
+      });
+    }
+  };
+
+  getAnchorPosition = () => {
+    const selection = document.getSelection();
+    const selectionRange = selection ? selection.getRangeAt(0) : null;
+    const selectionRect = selectionRange ? selectionRange.getBoundingClientRect() : null;
+    const selectionPositionY = selectionRect ? selectionRect.top : 0;
+    const selectionPositionX = selectionRect ? selectionRect.left : 0;
+    // $FlowFixMe this.fictionBodyView may be null
+    const bodyPosition = this.fictionBodyView.getBoundingClientRect().left - selectionRect.width / 2;
+    const anchorPositionX = selectionPositionX - bodyPosition - ANCHOR_OFFSET;
+    // $FlowFixMe this.fictionBodyView may be null
+    const anchorPositionY = selectionPositionY - this.fictionBodyView.getBoundingClientRect().top;
+    return { x: anchorPositionX, y: anchorPositionY };
+  };
+
+  handleMouseUpWhileHarvesting = () => {
```

Comment:
Maybe it's not relevant but what happen if the user is disconnected? Is it possible to harvest in this case?

Reply:
I ll had a check to only allow harvesting if connected